### ISSUE TITLE
nameservice client side transaction signing

### ISSUE THREAD

**Issue by tuckermint** [cosmos/sdk-tutorials#267]
Hi,

I made a wallet that works fine with https://api.cosmos.network/

But when I point it at the rest-server from a chain based on nameservice from commit 0e9f894fe4af3ae6c313614d80c262e6a53d3774, I get sequence / chain_id errors for all POST requests.

Is there some difference between Cosmos mainnet and this chain that requires a change to client side transaction signing?

I'm using https://github.com/cosmos/sig to sign the transactions.
My code is at https://github.com/tuckermint/tuckwallet
My wallet is hosted at http://tuckwallet.s3-website-us-east-1.amazonaws.com/

Thanks
@marbar3778

**Comment by tuckermint** [comment]
Or maybe there is some general difference between Cosmos SDK v0.36 and v0.38?

**Comment by tuckermint** [comment]
Here's the signed transaction produced by https://github.com/cosmos/sig
verifyTx returns true.
Using chain id "tuckermint", account number 517, sequence 2:
```
 {
  "msg": [
    {
      "type": "cosmos-sdk/MsgSend",
      "value": {
        "amount": [
          {
            "amount": "1000000",
            "denom": "utuckeratom"
          }
        ],
        "from_address": "cosmos18yk6fxjd2sgds5qk6zkhwjfdafxs8z535qyht9",
        "to_address": "cosmos12ff2ugqyj74cqyzzf8mjd332fjj0kwrwarjes9"
      }
    }
  ],
  "fee": {
    "amount": [
      {
        "denom": "utuckeratom",
        "amount": "5000"
      }
    ],
    "gas": "200000"
  },
  "memo": "TuckWallet",
  "signatures": [
    {
      "signature": "kqlFdzWWrJa+Z4jVxYXsRukfGzv75WVBGmV4PMWrRToH9ryJLjS0clqmB1Pja0OO6udVpoNKqxcEhVM9WnK80w==",
      "pub_key": {
        "type": "tendermint/PubKeySecp256k1",
        "value": "A1GfH7dB99vQxeMBS5HRYgYGeLk3+FfM9KFPqozenBT/"
      }
    }
  ]
}
```

--

Server side signed transaction with same inputs, created by this command:

nscli tx sign unsigned.json --from ACCOUNT --offline --chain-id tuckermint --sequence 2 --account-number 517 > signedTx.json

```
{
  "type": "cosmos-sdk/StdTx",
  "value": {
    "msg": [
      {
        "type": "cosmos-sdk/MsgSend",
        "value": {
          "from_address": "cosmos18yk6fxjd2sgds5qk6zkhwjfdafxs8z535qyht9",
          "to_address": "cosmos12ff2ugqyj74cqyzzf8mjd332fjj0kwrwarjes9",
          "amount": [
            {
              "denom": "utuckeratom",
              "amount": "1000000"
            }
          ]
        }
      }
    ],
    "fee": {
      "amount": [
        {
          "denom": "utuckeratom",
          "amount": "5000"
        }
      ],
      "gas": "200000"
    },
    "signatures": [
      {
        "pub_key": {
          "type": "tendermint/PubKeySecp256k1",
          "value": "A1GfH7dB99vQxeMBS5HRYgYGeLk3+FfM9KFPqozenBT/"
        },
        "signature": "JAj4hLW0QMuL0ncFoQOgIWEjqeCAvjUG6LBa8tVQ0B8BamEl+/85Qs2WfF4SlRrs8CXHGcPlako2yNNAkQMeOA=="
      }
    ],
    "memo": "TuckWallet"
  }
}
```
Is there a way to decode the signature to debug it?

Basically I don't know why Cosmos Sig and nscli tx sign are producing different signatures for the same transaction.

Also, I noticed that JSON.stringify puts the signature at the bottom of the transaction, while "nscli tx sign" puts the signature above the memo. Will this make a difference?

Also nscli tx sign makes me type my keyring passphrase twice. 

Thanks

**Comment by tuckermint** [comment]
@jordansexton @mircea-c

**Comment by marbar3778** [comment]
there is a difference between 0.36 and 0.38. Which are you using?

**Comment by tuckermint** [comment]
I tested my wallet on Cosmos mainnet, which I believe is 0.36
My custom chain is using 0.38, so I need to know how to modify my wallet to work with 0.38
The custom chain is based on nameservice from commit https://github.com/cosmos/sdk-tutorials/commit/0e9f894fe4af3ae6c313614d80c262e6a53d3774

**Comment by tuckermint** [comment]
The sign function from here produced the same signature as nscli: https://github.com/cybercongress/cyb-snap/blob/master/index.js

So I will use that
